### DISCUSSION TITLE
How to add cross-reference/label to the table

### DISCUSSION THREAD

**Discussion by dilawar** [ickc/pantable#46] [Q&A] [answered]
I use `pandoc-crossref` to generate cross-references. It works well.

I tried passing `#tbl:mytable` to code-fence e.g., `{#tbl:mytable .table}` and later in the main text, I tried referring to it by `@tbl:mytable`. This process failed to generate any `\label` for latex/pdf writer. I would have also liked to work for html writer (didn't try).

What is the recommended way to attach a label to the table which at least works for pdf and HTML?

**Comment by ghost** [comment]
That would be a very nice and important feature.

**Comment by ickc** [comment]
This might be an upstream problem in panflute. If I guess correctly from your description, after passing it to the filter, the attribute is lost. Can you provide a Minimal Working Example to reproduce this bug (including input file and command used to execute and its result)?

cc @sergiocorreia to see if he already know about missing attributes after filtering.

**Comment by sgelb** [comment, marked as answer]
Using [pandoc-crossref](https://lierdakil.github.io/pandoc-crossref/#table-labels), you have to add the label at the end of the caption:

```
---
caption: A nice caption {#tbl:captionlabel}
---
```

Reference it as usually:  `[@tbl:captionlabel]`

Do not forget to run the pantable filter before pandoc-crossref. This works for my using pantable 0.12.3, pandoc 2.9.2 and pandoc-crossref 0.3.6.3.

**Comment by ickc** [comment]
Thanks. Do you all think it would be beneficial to put in the doc or the GitHub wiki for future reference?

**Comment by sgelb** [comment]
I'm not sure. Pantable is not a tool for referencing, pandoc-crossref is. And they are very clear on how to add labels to captions, as you can see from the link from my previous comment. But on the other side, people did miss this and a at least a hint about the correct order if using other filters would certainly be an useful addition.
